### PR TITLE
Bugfix FXIOS-9758 ⁃ Bottom modal view(s) are not visible

### PR DESCRIPTION
**Pull request by dicarobinho**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9758)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21429)

## :bulb: Description
For iOS 16.0+ is used a custom detent in order to set the height for Tracking Protection Modal
Moved out from the scrollview, the header container view.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by cyndichin** [comment]
@dicarobinho Can you help clarify the design requirement here, I was unable to determine from the ticket. Want to see if there is a way to avoid the calculations and why not use the built-in detents.

**Comment by nbhasin2** [comment]
LGTM but @cyndichin feel free to double check

File: firefox-ios/Client/Extensions/UIButton+Extension.swift [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+extension UIButton {
+    func getHeight(with width: CGFloat) -> CGFloat {
+        return self.sizeThatFits(CGSize(width: width, height: .greatestFiniteMagnitude)).height
+    }
+}
```
Comment (cyndichin) [inline firefox-ios/Client/Extensions/UIButton+Extension.swift:11]: remove?

File: firefox-ios/Client/Extensions/UILabel+Extension.swift [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+extension UILabel {
+    func getHeight(with width: CGFloat) -> CGFloat {
+        return self.sizeThatFits(CGSize(width: width, height: .greatestFiniteMagnitude)).height
+    }
+}
```
Comment (cyndichin) [inline firefox-ios/Client/Extensions/UILabel+Extension.swift:11]: remove?

File: firefox-ios/Client/Extensions/UIButton+Extension.swift (hunk 1 above)
Comment (dicarobinho) [reply firefox-ios/Client/Extensions/UIButton+Extension.swift:11]: true

File: firefox-ios/Client/Extensions/UILabel+Extension.swift (hunk 2 above)
Comment (dicarobinho) [reply firefox-ios/Client/Extensions/UILabel+Extension.swift:11]: true

**Review by cyndichin — APPROVED** [review]
thanks for making the update, looks good! as mentioned in slack, would be good to communicate this to UX / QA as well as confirm how this works on iPad / iPhone for different versions

**Pull request merged**